docs: update AIO code to v15

bob-watson opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
The docs were configured for 15.0.0-rc.0.

## What is the new behavior?
The docs were configured for 15.0.0.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

Commit:
docs: update AIO code to v15

mary-poppins commented:
You can preview f2a9595 at https://pr48153-f2a9595.ngbuilds.io/.

AndrewKushnir reviewed (commented):
@bob-watson thanks for the update. The change itself looks good, but the AIO payload size is unexpected. We can still go ahead with this change and investigate later if needed.

bob-watson commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -2,7 +2,7 @@
     "uncompressed": {
       "runtime": 4325,
-      "main": 455143,
+      "main": 471785,
```
I don't have any idea where it grew.
I'm not in a particular hurry, but I wanted to get things sync'd up with v15 before I start writing [too many] more topics.

AndrewKushnir replied:
Got it, thanks. Let's see if @dgp1130 or @alan-agius4 may have some ideas on this size increase. Anyway, let's proceed with the change (to make sure no other work is blocked) and investigate the size increase as a followup.

alan-agius4 replied:
We should investigate this as a follow up for sure.

Mind creating a separate issue to keep track of this? I can probably look into this mid or later this week.

bob-watson replied:
https://github.com/angular/angular/issues/48171

clydin replied:
I spent some time this trying to investigate this.  I tried adding each package one at a time locally to try to reproduce the size increase but was unable to replicate the ~16KB change.  Also opened #48179 to test from CI and was also not able to replicate.  The sizes did go up a small amount (~600 bytes) though but i'm not sure why this PR has the large size change.

alan-agius4 replied:
The culprit is `safevalues`. I explain more about this in https://github.com/angular/angular/issues/48171#issuecomment-1324728758

AndrewKushnir reviewed (approved):
Reviewed-for: size-tracking.

mary-poppins commented:
You can preview 99d15d8 at https://pr48153-99d15d8.ngbuilds.io/.

bob-watson commented:
I created https://github.com/angular/angular/issues/48171 to follow up on this.

alan-agius4 committed:
build: update `safevalues` to `0.4.2`

This contains a fix which enabled the ESM version of the package to be consumed.

See: https://github.com/angular/angular/issues/48171#issuecomment-1324728758

mary-poppins commented:
You can preview ef74e83 at https://pr48153-ef74e83.ngbuilds.io/.

mary-poppins commented:
You can preview 1737c20 at https://pr48153-1737c20.ngbuilds.io/.
You can preview f2a9595 at https://pr48153-f2a9595.ngbuilds.io/.
You can preview 99d15d8 at https://pr48153-99d15d8.ngbuilds.io/.
You can preview ef74e83 at https://pr48153-ef74e83.ngbuilds.io/.

alan-agius4 commented:
@AndrewKushnir PTAL.

dylhunn reviewed (approved):
reviewed-for: size-tracking

alan-agius4 commented:
Caretaker note: note sure why the mergeablility status is not green.

AndrewKushnir commented:
@alan-agius4 @clydin huge thanks for the investigation, finding the root cause and the solution! 👍

dylhunn commented:
This PR was merged into the repository by commit 76e861f8dedd19258f1565df3b26d88cabe0c2a6.
